src/amount.cpp

```diff
@@ -12,14 +12,14 @@ const std::string CURRENCY_UNIT = "BTC";
 CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
 {
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid*1000/nSize;
+        nSatoshisPerK = nFeePaid * PRECISION_MULTIPLIER / nSize;
```

Comment:
Yeah, please let me follow up this with an "only for discussion". The name nSatoshisPerK is definitely part of what I don't like about CFeeRate and related to the point I want to make in that PR.

Maybe I should have left the constant ```PRECISION_MULTIPLIER = KB``` for that branch, but given that changing that  ```PRECISION_MULTIPLIER = KB+1``` will make the new test (among others) fail, I didn't saw how this makes the situation any worse.